In [6]:
import pandas as pd
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
# import glob
# import seaborn as sns
# %matplotlib notebook
%matplotlib inline
# from geopy.geocoders import Nominatim
# from geopy.extra.rate_limiter import RateLimiter

pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)
pd.set_option('display.width', 1000)

### Extracting Quebec subdivision levels

In [7]:
CensusDivisions = pd.read_csv('../Datasets/GeodataFiles/CensusDivisions/georef-canada-census-division.csv', sep=";", header=0)
CensusDivisionsQuebec = CensusDivisions[CensusDivisions['Official Code Province / Territory'] == 24].reset_index(drop=True)
CensusDivisionsQuebec

Geo Point                                          Geo Shape  Year  Official Code Province / Territory Official Name Province / Territory (English)  Official Code Census Division Official Name Census Division (English) Iso 3166-3 Area Code                    Type (English / French) Official Name Census Division (French) prov_name_fr
0   45.2144879226,-73.2297861079  {"type": "MultiPolygon", "coordinates": [[[[-7...  2019                                  24                                       Quebec                           2456                       Le Haut-Richelieu                  CAN            Municipalité régionale de comté                      Le Haut-Richelieu       Québec
1   45.9994319791,-71.9430644096  {"type": "Polygon", "coordinates": [[[-72.3050...  2019                                  24                                       Quebec                           2439                              Arthabaska                  CAN            Municipalité régionale de comté                             Arthabaska       Québec
2   45.8285967363,-73.4472570549  {"type": "MultiPolygon", "coordinates": [[[[-7...  2019                                  24                                       Quebec                           2460                            L'Assomption                  CAN            Municipalité régionale de comté                           L'Assomption       Québec
3   45.5357751757,-74.0353053841  {"type": "Polygon", "coordinates": [[[-74.2785...  2019                                  24                                       Quebec                           2472                          Deux-Montagnes                  CAN            Municipalité régionale de comté                         Deux-Montagnes       Québec
4   45.2358950468,-73.9434723433  {"type": "MultiPolygon", "coordinates": [[[[-7...  2019                                  24                                       Quebec                           2470                   Beauharnois-Salaberry                  CAN            Municipalité régionale de comté                  Beauharnois-Salaberry       Québec
5   48.4467134206,-67.0889326544  {"type": "Polygon", "coordinates": [[[-66.2798...  2019                                  24                                       Quebec                           2407                            La Matapédia                  CAN            Municipalité régionale de comté                           La Matapédia       Québec
6   51.0744600084,-61.9296506678  {"type": "MultiPolygon", "coordinates": [[[[-6...  2019                                  24                                       Quebec                           2498     Minganie--Le Golfe-du-Saint-Laurent                  CAN  Census division / Division de recensement    Minganie--Le Golfe-du-Saint-Laurent       Québec
7   46.4253334093,-77.0937060359  {"type": "Polygon", "coordinates": [[[-76.1509...  2019                                  24                                       Quebec                           2484                                 Pontiac                  CAN            Municipalité régionale de comté                                Pontiac       Québec
8   46.0499382693,-73.4527647018  {"type": "Polygon", "coordinates": [[[-73.6102...  2019                                  24                                       Quebec                           2461                                Joliette                  CAN            Municipalité régionale de comté                               Joliette       Québec
9   45.7059762959,-73.8327771953  {"type": "Polygon", "coordinates": [[[-73.8240...  2019                                  24                                       Quebec                           2473                   Thérèse-De Blainville                  CAN            Municipalité régionale de comté                  Thérèse-De Blainville       Québec
10  49.0807765361,-69.5685863374  {"type": "Polygon", "coordinates": [[[-70.0256...  

### DGUID:
https://www150.statcan.gc.ca/t1/tbl1/en/tv.action?pid=3210037001

We will consider the most recent data (2021) for the census divisions of Quebec:

Vintage: 2021

Type: A

Schema: 0003

Geographic Unique Identifier (GEOUID): Previous dataset (georef-canada-census-division.csv)

In [8]:
DGUID_Quebec = pd.DataFrame()

DGUID_Vintage = '2021'

DGUID_Type = 'A'

DGUID_Schema =  '0003'

DGUID_Quebec['DGUID'] = DGUID_Vintage+DGUID_Type+DGUID_Schema+CensusDivisionsQuebec['Official Code Census Division'].astype(str)
DGUID_Quebec

DGUID
0   2021A00032456
1   2021A00032439
2   2021A00032460
3   2021A00032472
4   2021A00032470
5   2021A00032407
6   2021A00032498
7   2021A00032484
8   2021A00032461
9   2021A00032473
10  2021A00032495
11  2021A00032496
12  2021A00032412
13  2021A00032463
14  2021A00032418
15  2021A00032445
16  2021A00032437
17  2021A00032450
18  2021A00032476
19  2021A00032481
20  2021A00032479
21  2021A00032477
22  2021A00032452
23  2021A00032491
24  2021A00032458
25  2021A00032416
26  2021A00032415
27  2021A00032425
28  2021A00032406
29  2021A00032485
30  2021A00032465
31  2021A00032403
32  2021A00032446
33  2021A00032408
34  2021A00032441
35  2021A00032497
36  2021A00032440
37  2021A00032499
38  2021A00032422
39  2021A00032464
40  2021A00032448
41  2021A00032490
42  2021A00032454
43  2021A00032421
44  2021A00032404
45  2021A00032438
46  2021A00032426
47  2021A00032444
48  2021A00032429
49  2021A00032433
50  2021A00032459
51  2021A00032474
52  2021A00032430
53  2021A00032469
54  2021A00032455
55  2021A00032410
56  2021A00032457
57  2021A00032494
58  2021A00032420
59  2021A00032482
60  2021A00032413
61  2021A00032468
62  2021A00032411
63  2021A00032449
64  2021A00032417
65  2021A00032467
66  2021A00032486
67  2021A00032489
68  2021A00032409
69  2021A00032402
70  2021A00032431
71  2021A00032401
72  2021A00032466
73  2021A00032462
74  2021A00032487
75  2021A00032432
76  2021A00032493
77  2021A00032435
78  2021A00032492
79  2021A00032427
80  2021A00032436
81  2021A00032488
82  2021A00032453
83  2021A00032414
84  2021A00032483
85  2021A00032442
86  2021A00032447
87  2021A00032471
88  2021A00032478
89  2021A00032419
90  2021A00032428
91  2021A00032405
92  2021A00032434
93  2021A00032443
94  2021A00032451
95  2021A00032423
96  2021A00032475
97  2021A00032480

### Extracting Quebec subdivision levels

In [9]:
CensusDivisions = pd.read_csv('../Datasets/GeodataFiles/CensusDivisions/georef-canada-census-division.csv', sep=";", header=0)
CensusDivisionsQuebec = CensusDivisions[CensusDivisions['Official Code Province / Territory'] == 24].reset_index(drop=True)
# CensusDivisionsQuebec['Official Code Census Division']
CensusDivisionsQuebec

Geo Point                                          Geo Shape  Year  Official Code Province / Territory Official Name Province / Territory (English)  Official Code Census Division Official Name Census Division (English) Iso 3166-3 Area Code                    Type (English / French) Official Name Census Division (French) prov_name_fr
0   45.2144879226,-73.2297861079  {"type": "MultiPolygon", "coordinates": [[[[-7...  2019                                  24                                       Quebec                           2456                       Le Haut-Richelieu                  CAN            Municipalité régionale de comté                      Le Haut-Richelieu       Québec
1   45.9994319791,-71.9430644096  {"type": "Polygon", "coordinates": [[[-72.3050...  2019                                  24                                       Quebec                           2439                              Arthabaska                  CAN            Municipalité régionale de comté                             Arthabaska       Québec
2   45.8285967363,-73.4472570549  {"type": "MultiPolygon", "coordinates": [[[[-7...  2019                                  24                                       Quebec                           2460                            L'Assomption                  CAN            Municipalité régionale de comté                           L'Assomption       Québec
3   45.5357751757,-74.0353053841  {"type": "Polygon", "coordinates": [[[-74.2785...  2019                                  24                                       Quebec                           2472                          Deux-Montagnes                  CAN            Municipalité régionale de comté                         Deux-Montagnes       Québec
4   45.2358950468,-73.9434723433  {"type": "MultiPolygon", "coordinates": [[[[-7...  2019                                  24                                       Quebec                           2470                   Beauharnois-Salaberry                  CAN            Municipalité régionale de comté                  Beauharnois-Salaberry       Québec
5   48.4467134206,-67.0889326544  {"type": "Polygon", "coordinates": [[[-66.2798...  2019                                  24                                       Quebec                           2407                            La Matapédia                  CAN            Municipalité régionale de comté                           La Matapédia       Québec
6   51.0744600084,-61.9296506678  {"type": "MultiPolygon", "coordinates": [[[[-6...  2019                                  24                                       Quebec                           2498     Minganie--Le Golfe-du-Saint-Laurent                  CAN  Census division / Division de recensement    Minganie--Le Golfe-du-Saint-Laurent       Québec
7   46.4253334093,-77.0937060359  {"type": "Polygon", "coordinates": [[[-76.1509...  2019                                  24                                       Quebec                           2484                                 Pontiac                  CAN            Municipalité régionale de comté                                Pontiac       Québec
8   46.0499382693,-73.4527647018  {"type": "Polygon", "coordinates": [[[-73.6102...  2019                                  24                                       Quebec                           2461                                Joliette                  CAN            Municipalité régionale de comté                               Joliette       Québec
9   45.7059762959,-73.8327771953  {"type": "Polygon", "coordinates": [[[-73.8240...  2019                                  24                                       Quebec                           2473                   Thérèse-De Blainville                  CAN            Municipalité régionale de comté                  Thérèse-De Blainville       Québec
10  49.0807765361,-69.5685863374  {"type": "Polygon", "coordinates": [[[-70.0256...  

# Milk

## Milk production for whole Quebec (Milk production statistics, by administrative region, regional county municipality (RCM) and for all of Québec, https://statistique.quebec.ca/en/document/milk-production-statistics-by-administrative-region-regional-county-municipality-rcm-and-for-all-of-quebec) and assigned to each Census Division based on the number of dairy cows reported by the Census of agriculture

In [28]:
Milk_Quebec = 33478860*100 #liters

#Import animals in CDs from manure files. Dairy heifers do not produce milk
CattleManureQuebec = pd.read_csv('DataConditioning/'+'Cattle'+'_'+'Manure'+'.csv', sep=",", header=0)

DairyCowQuebec = CattleManureQuebec[['CD','DairyCow', 'DairyCow_AU']]

DairyCowTotal = DairyCowQuebec.sum(numeric_only=True)['DairyCow']
DairyCowAUTotal = DairyCowQuebec.sum(numeric_only=True)['DairyCow_AU']

#Share of cows in each CD
DairyCowQuebec['% of dairy cows (animals) in CD over total Quebec'] = DairyCowQuebec['DairyCow']/DairyCowTotal*100
DairyCowQuebec['% of dairy cows (AU) in CD over total Quebec'] = DairyCowQuebec['DairyCow_AU']/DairyCowAUTotal*100

# Assingning milk based on cows
DairyCowQuebec['Milk (liters)'] = DairyCowQuebec['% of dairy cows (animals) in CD over total Quebec']/100*Milk_Quebec

DairyCowQuebec

/tmp/ipykernel_54410/1919803277.py:12: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DairyCowQuebec['% of dairy cows (animals) in CD over total Quebec'] = DairyCowQuebec['DairyCow']/DairyCowTotal*100
/tmp/ipykernel_54410/1919803277.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DairyCowQuebec['% of dairy cows (AU) in CD over total Quebec'] = DairyCowQuebec['DairyCow_AU']/DairyCowAUTotal*100
/tmp/ipykernel_54410/1919803277.py:16: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice

CD  DairyCow   DairyCow_AU  % of dairy cows (animals) in CD over total Quebec  % of dairy cows (AU) in CD over total Quebec  Milk (liters)
0   2401      90.0    121.621622                                           0.025338                                      0.025338   8.482915e+05
1   2402       0.0      0.000000                                           0.000000                                      0.000000   0.000000e+00
2   2403       0.0      0.000000                                           0.000000                                      0.000000   0.000000e+00
3   2404       0.0      0.000000                                           0.000000                                      0.000000   0.000000e+00
4   2405     170.0    229.729730                                           0.047861                                      0.047861   1.602328e+06
5   2406     462.0    624.324324                                           0.130069                                      0.130069   4.354563e+06
6   2407    3615.0   4885.135135                                           1.017748                                      1.017748   3.407304e+07
7   2408    1992.0   2691.891892                                           0.560817                                      0.560817   1.877552e+07
8   2409    3873.0   5233.783784                                           1.090384                                      1.090384   3.650481e+07
9   2410    5509.0   7444.594595                                           1.550975                                      1.550975   5.192486e+07
10  2411    3354.0   4532.432432                                           0.944267                                      0.944267   3.161300e+07
11  2412    6240.0   8432.432432                                           1.756777                                      1.756777   5.881488e+07
12  2413    2810.0   3797.297297                                           0.791113                                      0.791113   2.648555e+07
13  2414   12667.0  17117.567568                                           3.566200                                      3.566200   1.193923e+08
14  2415     295.0    398.648649                                           0.083053                                      0.083053   2.780511e+06
15  2416     455.0    614.864865                                           0.128098                                      0.128098   4.288585e+06
16  2417    4844.0   6545.945946                                           1.363754                                      1.363754   4.565693e+07
17  2418    4506.0   6089.189189                                           1.268595                                      1.268595   4.247113e+07
18  2419   15818.0  21375.675676                                           4.453316                                      4.453316   1.490919e+08
19  2420     893.0   1206.756757                                           0.251410                                      0.251410   8.416937e+06
20  2421     307.0    414.864865                                           0.086431                                      0.086431   2.893616e+06
21  2422       0.0      0.000000                                           0.000000                                      0.000000   0.000000e+00
22  2423       0.0      0.000000                                           0.000000                                      0.000000   0.000000e+00
23  2425     957.0   1293.243243                                           0.269429                                      0.269429   9.020166e+06
24  2426   10863.0  14679.729730                                           3.058311                                      3.058311   1.023888e+08
25  2427    8065.0  10898.648649                                           2.270577                                      2.270577   7.601634e+07
26  2428    1138.0   1537.837838                                           0.320386        

### P contained in milk

0.94 g P per L of milk (Health Canada, 2008)

https://www.canada.ca/content/dam/hc-sc/migration/hc-sc/fn-an/alt_formats/pdf/nutrition/fiche-nutri-data/nvscf-vnqau-eng.pdf

In [29]:
P_Milk = 0.94E-3 #kg per L

In [30]:
DairyCowQuebec['Milk (kg P)'] = DairyCowQuebec['Milk (liters)']*P_Milk

DairyCowQuebec.to_csv('../Results/AgriProducts/P_MilkQuebec.csv')

DairyCowQuebec

/tmp/ipykernel_54410/2954193954.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  DairyCowQuebec['Milk (kg P)'] = DairyCowQuebec['Milk (liters)']*P_Milk


CD  DairyCow   DairyCow_AU  % of dairy cows (animals) in CD over total Quebec  % of dairy cows (AU) in CD over total Quebec  Milk (liters)    Milk (kg P)
0   2401      90.0    121.621622                                           0.025338                                      0.025338   8.482915e+05     797.393990
1   2402       0.0      0.000000                                           0.000000                                      0.000000   0.000000e+00       0.000000
2   2403       0.0      0.000000                                           0.000000                                      0.000000   0.000000e+00       0.000000
3   2404       0.0      0.000000                                           0.000000                                      0.000000   0.000000e+00       0.000000
4   2405     170.0    229.729730                                           0.047861                                      0.047861   1.602328e+06    1506.188647
5   2406     462.0    624.324324                                           0.130069                                      0.130069   4.354563e+06    4093.289148
6   2407    3615.0   4885.135135                                           1.017748                                      1.017748   3.407304e+07   32028.658590
7   2408    1992.0   2691.891892                                           0.560817                                      0.560817   1.877552e+07   17648.986974
8   2409    3873.0   5233.783784                                           1.090384                                      1.090384   3.650481e+07   34314.521361
9   2410    5509.0   7444.594595                                           1.550975                                      1.550975   5.192486e+07   48809.372109
10  2411    3354.0   4532.432432                                           0.944267                                      0.944267   3.161300e+07   29716.216020
11  2412    6240.0   8432.432432                                           1.756777                                      1.756777   5.881488e+07   55285.983293
12  2413    2810.0   3797.297297                                           0.791113                                      0.791113   2.648555e+07   24896.412348
13  2414   12667.0  17117.567568                                           3.566200                                      3.566200   1.193923e+08  112228.774097
14  2415     295.0    398.648649                                           0.083053                                      0.083053   2.780511e+06    2613.680300
15  2416     455.0    614.864865                                           0.128098                                      0.128098   4.288585e+06    4031.269615
16  2417    4844.0   6545.945946                                           1.363754                                      1.363754   4.565693e+07   42917.516518
17  2418    4506.0   6089.189189                                           1.268595                                      1.268595   4.247113e+07   39922.859089
18  2419   15818.0  21375.675676                                           4.453316                                      4.453316   1.490919e+08  140146.423673
19  2420     893.0   1206.756757                                           0.251410                                      0.251410   8.416937e+06    7911.920365
20  2421     307.0    414.864865                                           0.086431                                      0.086431   2.893616e+06    2719.999499
21  2422       0.0      0.000000                                           0.000000                                      0.000000   0.000000e+00       0.000000
22  2423       0.0      0.000000                                           0.000000                                      0.000000   0.000000e+00       0.000000
23  2425     957.0   1293.243243                                           0.269429                                      0.269429   9.020166e+06    8478.956092
24  24

# Eggs


In [39]:
Eggs = pd.read_csv('../Datasets/AgriProductsProduction/Eggs/32100376.csv', sep=",", header=0)

EggsQuebec = DGUID_Quebec.merge(Eggs, how='inner', on='DGUID')

EggsQuebec['CD'] = EggsQuebec['DGUID'].str.split('A0003').str[1]

EggsQuebec = EggsQuebec[(EggsQuebec['Unit of measure'] == 'Number of eggs (dozens)') & (EggsQuebec['Egg production'] == 'Table eggs')]

EggsQuebec = EggsQuebec.fillna(0)

EggsQuebec = EggsQuebec.rename(columns={'VALUE':'Eggs (dozens)'})

EggsQuebec['Eggs (number)'] = EggsQuebec['Eggs (dozens)']*12


EggsQuebec

DGUID  REF_DATE                                                GEO Egg production          Unit of measure     UOM  UOM_ID SCALAR_FACTOR  SCALAR_ID       VECTOR COORDINATE  Eggs (dozens) STATUS  SYMBOL  TERMINATED  DECIMALS    CD  Eggs (number)
1    2021A00032456      2021            Le Haut-Richelieu, Quebec [CD241356000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327135800    905.1.2            0.0      F     0.0         0.0         0  2456            0.0
5    2021A00032439      2021                   Arthabaska, Quebec [CD241439000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327136180   1000.1.2            0.0      F     0.0         0.0         0  2439            0.0
9    2021A00032460      2021                 L'Assomption, Quebec [CD240760000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327134316    534.1.2      1813280.0      D     0.0         0.0         0  2460     21759360.0
13   2021A00032472      2021               Deux-Montagnes, Quebec [CD240972000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327134640    615.1.2            0.0      F     0.0         0.0         0  2472            0.0
17   2021A00032470      2021        Beauharnois-Salaberry, Quebec [CD241370000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327136028    962.1.2            0.0      F     0.0         0.0         0  2470            0.0
21   2021A00032407      2021                 La Matapédia, Quebec [CD240107000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327133180    250.1.2      1278332.0      D     0.0         0.0         0  2407     15339984.0
25   2021A00032498      2021  Minganie--Le Golfe-du-Saint-Laurent, Quebec [C...     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327133656    369.1.2            0.0      A     0.0         0.0         0  2498            0.0
29   2021A00032484      2021                      Pontiac, Quebec [CD240884000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327134596    604.1.2            0.0      F     0.0         0.0         0  2484            0.0
33   2021A00032461      2021                     Joliette, Quebec [CD240761000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327134332    538.1.2            0.0      F     0.0         0.0         0  2461            0.0
37   2021A00032473      2021        Thérèse-De Blainville, Quebec [CD240973000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327134660    620.1.2            0.0      F     0.0         0.0         0  2473            0.0
41   2021A00032495      2021           La Haute-Côte-Nord, Quebec [CD240295000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327133624    361.1.2            0.0      F     0.0         0.0         0  2495            0.0
45   2021A00032496      2021                  Manicouagan, Quebec [CD240296000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327133640    365.1.2            0.0      F     0.0         0.0         0  2496            0.0
49   2021A00032412      2021              Rivière-du-Loup, Quebec [CD240112000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327133344    291.1.2            0.0      F     0.0         0.0         0  2412            0.0
53   2021A00032463      2021                     Montcalm, Quebec [CD240763000]     Table eggs  Number of eggs (dozens)  Number     223         units          0  v1327134412    558.1.2      2209509.0      E     0.0         0.0         0  2463     26514108.0
57   2021A00032418      2021                    Montmagny, Quebec [CD241218000]     Table eggs  Number of eggs (dozens)  Number     223        

### P contained in egg

59.5 g per egg (https://www.sciencedirect.com/science/article/pii/B9780128008799000019)

170mg P per 100 g of egg (Standardised calculation methods for animal manure and nutrients) THIS IS THE ONE USED; NOT THE ONES BELOW
    
+ 173.58 http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.1056.3102&rep=rep1&type=pdf 
    
+ 158.17 http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.1056.3102&rep=rep1&type=pdf 
    
+ 173 https://www.semanticscholar.org/paper/Effect-of-housing-systems-on-biochemical-of-chicken-Matt-Veromann/92075e20533734bde682b11d9abea77b2c679737  
    
+ 164 https://www.semanticscholar.org/paper/Effect-of-housing-systems-on-biochemical-of-chicken-Matt-Veromann/92075e20533734bde682b11d9abea77b2c679737 

+ 198 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6470839/ 

In [44]:
Egg_Mass = 59.5 #g/egg
PinEgg = (((170)*1E-3)/100*Egg_Mass)/1E3 #kg in egg

In [47]:
EggsQuebec_summary = EggsQuebec[['DGUID','CD','Eggs (number)']]

EggsQuebec_summary['Eggs (kg P)'] = EggsQuebec_summary['Eggs (number)'] * PinEgg

EggsQuebec_summary.to_csv('../Results/AgriProducts/P_EggsQuebec.csv')


EggsQuebec_summary

/tmp/ipykernel_54410/3868413417.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  EggsQuebec_summary['Eggs (kg P)'] = EggsQuebec_summary['Eggs (number)'] * PinEgg


DGUID    CD  Eggs (number)   Eggs (kg P)
1    2021A00032456  2456            0.0      0.000000
5    2021A00032439  2439            0.0      0.000000
9    2021A00032460  2460     21759360.0   2200.959264
13   2021A00032472  2472            0.0      0.000000
17   2021A00032470  2470            0.0      0.000000
21   2021A00032407  2407     15339984.0   1551.639382
25   2021A00032498  2498            0.0      0.000000
29   2021A00032484  2484            0.0      0.000000
33   2021A00032461  2461            0.0      0.000000
37   2021A00032473  2473            0.0      0.000000
41   2021A00032495  2495            0.0      0.000000
45   2021A00032496  2496            0.0      0.000000
49   2021A00032412  2412            0.0      0.000000
53   2021A00032463  2463     26514108.0   2681.902024
57   2021A00032418  2418            0.0      0.000000
61   2021A00032445  2445            0.0      0.000000
65   2021A00032437  2437            0.0      0.000000
69   2021A00032450  2450    112521504.0  11381.550130
73   2021A00032476  2476            0.0      0.000000
77   2021A00032481  2481            0.0      0.000000
81   2021A00032479  2479     13083372.0   1323.383078
85   2021A00032477  2477            0.0      0.000000
89   2021A00032452  2452     15904992.0   1608.789941
93   2021A00032491  2491            0.0      0.000000
97   2021A00032458  2458            0.0      0.000000
101  2021A00032416  2416     15001848.0   1517.436925
105  2021A00032415  2415            0.0      0.000000
109  2021A00032425  2425            0.0      0.000000
113  2021A00032406  2406            0.0      0.000000
117  2021A00032485  2485            0.0      0.000000
121  2021A00032465  2465            0.0      0.000000
125  2021A00032403  2403            0.0      0.000000
129  2021A00032446  2446    152909148.0  15466.760320
133  2021A00032408  2408            0.0      0.000000
137  2021A00032441  2441            0.0      0.000000
141  2021A00032497  2497            0.0      0.000000
145  2021A00032440  2440            0.0      0.000000
149  2021A00032499  2499            0.0      0.000000
153  2021A00032422  2422            0.0      0.000000
157  2021A00032464  2464            0.0      0.000000
161  2021A00032448  2448            0.0      0.000000
165  2021A00032490  2490            0.0      0.000000
169  2021A00032454  2454    290082876.0  29341.882907
173  2021A00032421  2421      4639860.0    469.321839
177  2021A00032404  2404            0.0      0.000000
181  2021A00032438  2438            0.0      0.000000
185  2021A00032426  2426    129636384.0  13112.720242
189  2021A00032444  2444            0.0      0.000000
193  2021A00032429  2429            0.0      0.000000
197  2021A00032433  2433            0.0      0.000000
201  2021A00032459  2459            0.0      0.000000
205  2021A00032474  2474            0.0      0.000000
209  2021A00032430  2430            0.0      0.000000
213  2021A00032469  2469            0.0      0.000000
217  2021A00032455  2455    107136240.0  10836.830676
221  2021A00032410  2410            0.0      0.000000
225  2021A00032457  2457     39966264.0   4042.587604
229  2021A00032494  2494     15907644.0   1609.058191
233  2021A00032420  2420      5391600.0    545.360340
237  2021A00032482  2482            0.0      0.000000
241  2021A00032413  2413            0.0      0.000000
245  2021A00032468  2468            0.0      0.000000
249  2021A00032411  2411            0.0      0.000000
253  2021A00032449  2449    230979408.0  23363.567119
257  2021A00032417  2417            0.0      0.000000
261  2021A00032467  2467            0.0      0.000000
265  2021A00032486  2486            0.0      0.000000
269  2021A00032489  2489     50809800.0   5139.411270
273  2021A00032409  2409            0.0      0.000000
277  2021A00032402  2402            0.0      0.000000
281  2021A00032431  2431            0.0      0.000000
285  2021A00032401  2401      2919996.0    295.357595
289  2021A00032466  2466            0.0      0.000000
293  2021A0003246

# Meat

For rabbits we took the percentaje over the total rabbits slaughtered in Canada for 2019 (last year available) reported by provincial governemtn (19%, to account for a new facility oppened in 2019) (Portrait-diagnostic sectoriel de l'industrie cunicole au Québec
 https://www.mapaq.gouv.qc.ca/fr/md/Publications/Pages/Details-Publication.aspx?guid=%7b389dc98d-89de-4873-956b-37486ade2a04%7d) and applied to the total number of rabbit reported by the federal government for 2021 (https://agriculture.canada.ca/en/sector/animal-industry/red-meat-and-livestock-market-information/rabbit-industry-glance)

For poultry other than chicken, Quebec and the atlantic provinces are rported together. The data was broken down using the % of chicken slaughtered in Quebec regarding the total chicken slaughted in the sum of Quebec and the atlantic provinces (71.9%)

For cattle, the number of animals slaughtered in federal slaughterhouses is provided as the sum of Quebec and the atlantic provinces. For breaking the data down, the % of animals sluaghtered in ferderal slaughterhoused over the total of animals slaughtered in Quebec for 2019 (last year available) is taken from a provincial report (for calves 92.3% are federal and 7.7% provincial, and for cows 46.1% are federal and 53.9% are provincial), and then applied to the animals slaugthered in the provincial plants of Quebec 

In [36]:
SlaughterHogsTotal = pd.read_csv('../Datasets/AgriProductsProduction/Meat/SlaughterHogsTotal_Conditioned.csv', sep=",", header=0)
SlaughterHogsTotal = SlaughterHogsTotal.rename(columns={'Unnamed: 0':'Animals'}).set_index('Animals')

SlaughterSheepTotal = pd.read_csv('../Datasets/AgriProductsProduction/Meat/SlaughterSheepTotal.csv', sep=",", header=0)
SlaughterSheepTotal = SlaughterSheepTotal.rename(columns={'Unnamed: 0':'Animals'}).set_index('Animals')
# SlaughterSheepTotal = SlaughterSheepTotal_Conditioned.loc['ProvincialSheep']

SlaughterGoatTotal = pd.read_csv('../Datasets/AgriProductsProduction/Meat/SlaughterGoatsTotal.csv', sep=",", header=0).set_index('Provinces')

SlaughterCalvesTotal = pd.read_csv('../Datasets/AgriProductsProduction/Meat/SlaughterCalvesTotal_Conditioned.csv', sep=",", header=0)
SlaughterCalvesTotal = SlaughterCalvesTotal.rename(columns={'Unnamed: 0':'Animals'}).set_index('Animals')

SlaughterCattleTotal = pd.read_csv('../Datasets/AgriProductsProduction/Meat/SlaughterCattleTotal_Conditioned.csv', sep=",", header=0)
SlaughterCattleTotal = SlaughterCattleTotal.rename(columns={'Unnamed: 0':'Animals'}).set_index('Animals')

SlaughterChickenTotal = pd.read_csv('../Datasets/AgriProductsProduction/Meat/Chicken_Total_Conditioned.csv', sep=",", header=0)
SlaughterChickenTotal = SlaughterChickenTotal.set_index('Chicken')
SlaughterChickenTotal = SlaughterChickenTotal.transpose()
SlaughterChickenTotal = SlaughterChickenTotal[sorted(SlaughterChickenTotal.columns.to_list())]
SlaughterChickenTotal = SlaughterChickenTotal.rename(columns=dict(zip(SlaughterChickenTotal.columns.to_list(),SlaughterChickenTotal.columns.astype(str).to_list())))
SlaughterChickenTotal.loc['ChickenWeight'] = SlaughterChickenTotal.loc['Kg']/SlaughterChickenTotal.loc['Head']
# SlaughterChickenTotal = SlaughterChickenTotal['2021']

SlaughterTurkeyTotal = pd.read_csv('../Datasets/AgriProductsProduction/Meat/Turkey_Total_Conditioned.csv', sep=",", header=0)
SlaughterTurkeyTotal = SlaughterTurkeyTotal.set_index('Turkey')
SlaughterTurkeyTotal = SlaughterTurkeyTotal.transpose()
SlaughterTurkeyTotal = SlaughterTurkeyTotal[sorted(SlaughterTurkeyTotal.columns.to_list())]
SlaughterTurkeyTotal = SlaughterTurkeyTotal.rename(columns=dict(zip(SlaughterTurkeyTotal.columns.to_list(),SlaughterTurkeyTotal.columns.astype(str).to_list())))
SlaughterTurkeyTotal.loc['TurkeyWeight'] = SlaughterTurkeyTotal.loc['Kg']/SlaughterTurkeyTotal.loc['Head']

SlaughterRabbitsTotal = pd.read_csv('../Datasets/AgriProductsProduction/Meat/white_meat_RabbitProvincial_Conditioned.csv', sep=",", header=0)
SlaughterRabbitsTotal = SlaughterRabbitsTotal.rename(columns={'Unnamed: 0':'Year'}).set_index('Year')

SlaughterChickenTotal

Chicken                2020          2021
Head           1.824764e+08  1.857100e+08
Kg             4.335044e+08  4.411944e+08
ChickenWeight  2.375673e+00  2.375716e+00

In [40]:
AnimalsTotal = pd.DataFrame(columns=['2021'])
AnimalsTotal.loc['Hogs'] = SlaughterHogsTotal.loc['HogSlaughtered_Total']['2021']
AnimalsTotal.loc['Sheep'] = SlaughterSheepTotal.loc['ProvincialSheep']['2021']
AnimalsTotal.loc['Goat'] = SlaughterGoatTotal.loc['Quebec']['2021']
AnimalsTotal.loc['Calves'] = SlaughterCalvesTotal.loc['CalvesSlaughtered_Total']['2021']
AnimalsTotal.loc['Cattle'] = SlaughterCattleTotal.loc['CattleSlaughtered_Total']['2021']
AnimalsTotal.loc['Chicken'] = SlaughterChickenTotal.loc['Head']['2021']
AnimalsTotal.loc['Turkey'] = SlaughterTurkeyTotal.loc['Head']['2021']
AnimalsTotal.loc['Rabbit'] = SlaughterRabbitsTotal.loc['Number of Heads']['2021']

AnimalsTotal = AnimalsTotal.rename(columns=dict(zip(AnimalsTotal.columns.to_list(),[i+' (heads)' for i in AnimalsTotal.columns.to_list()])))
AnimalsTotal.index.name = 'Animals'

AnimalsTotal.to_csv('../Datasets/AgriProductsProduction/Meat/Conditioned/AnimalsTotal_Conditoned.csv')

AnimalsTotal

2021 (heads)
Animals              
Hogs     6.932292e+06
Sheep    7.551700e+04
Goat     5.757000e+03
Calves   1.529740e+05
Cattle   4.330241e+04
Chicken  1.857100e+08
Turkey   3.430198e+06
Rabbit   9.016700e+04

### Breaking down data into census division level

Considerations:

15% of total livestock is processed in provincially inspected facilities. It is assumed that 100% of sheep, lambs, goats, and rabbits are processed in these faciltities (https://www.attorneygeneral.jus.gov.on.ca/english/about/pubs/meatinspectionreport/Chapter13.pdf)

The remaining 85% of total livestock, comprised of cattle, swine, and poultry, is processed in federally inspected facilities.(https://www.attorneygeneral.jus.gov.on.ca/english/about/pubs/meatinspectionreport/Chapter13.pdf)

NOT Federally inspected facilities are geolocated based on the data provided by the Canadian Food Inspection Agency (https://inspection.canada.ca/active/scripts/meavia/reglist/reglist.asp) by using Nominatim and OpenStreetMaps. Only facilities located in Ontario reporting slaughteirng activity are considered.

NOT Geolocation of provincially inspected facilities is not provided by OMAFRA (https://data.ontario.ca/en/dataset/provincially-licensed-meat-plants). 

Therefore, the animals slaughtered in each census division are estimated based on the emplyoment data for meat processing plants reported by the Canadian Business Counts report for each year under evaluation since 2018 (because of this is the first year reported at census division level (https://www150.statcan.gc.ca/t1/tbl1/en/tv.action?pid=3310039701&pickMembers%5B0%5D=3.168). The NAICS code for slaughtering is the 311611, but 3116 (meat processing) havs been used

In [49]:
EmployeesMeatPlants = pd.read_csv('../Datasets/AgriProductsProduction/Meat/EmployeesMeatPlants/2021/33100397-eng/33100397.csv', sep=",", header=0)

# Filtering for Meat product manufacturing [3116] activities
EmployeesMeatPlants = EmployeesMeatPlants[EmployeesMeatPlants['North American Industry Classification System (NAICS)'].str.contains("3116")==True]

# Filtering for Quebec facilities
DGUID_Vintage = '2016' #Someone in the goverment forgot to upodate the vintage of the file
DGUID_Type = 'A'
DGUID_Schema =  '0005' #The data is reporte at census subdiviosn
DGUID_Quebec = '24'
DGUID_filter = DGUID_Vintage+DGUID_Type+DGUID_Schema+DGUID_Quebec

EmployeesMeatPlants_Quebec = EmployeesMeatPlants[EmployeesMeatPlants['DGUID'].str.contains(DGUID_filter)==True]

# Removing rows Total, with employees
EmployeesMeatPlants_Quebec = EmployeesMeatPlants_Quebec[~EmployeesMeatPlants_Quebec['Employment size'].str.contains("Total, with employees")]

# Obtaining  CD
EmployeesMeatPlants_Quebec = EmployeesMeatPlants_Quebec.reset_index(drop=True)
EmployeesMeatPlants_Quebec['CD'] = [i.split(DGUID_Vintage+DGUID_Type+DGUID_Schema)[1][0:4] for i in EmployeesMeatPlants_Quebec['DGUID'].to_list()]

#Estimating employees
EmployeesDict = {'1 to 4 employees':2.5,
                     '5 to 9 employees':7,
                        '10 to 19 employees':14.5,
                        '20 to 49 employees':34.5,
                        '50 to 99 employees':74.5,
                        '100 to 199 employees':149.5,
                        '200 to 499 employees':349.5,
                        '500 plus employees':700,
                    }

EmployeesMeatPlants_Quebec['Employees'] = np.nan
for index, row in EmployeesMeatPlants_Quebec.iterrows():
    EmployeesMeatPlants_Quebec.loc[index, 'Employees'] = EmployeesDict[EmployeesMeatPlants_Quebec.loc[index, 'Employment size']]*EmployeesMeatPlants_Quebec.loc[index,'VALUE'].astype(int)

# Grouping by Census Division
EmployeesMeatPlants_Quebec = EmployeesMeatPlants_Quebec.groupby(['CD']).sum(numeric_only=True).drop(['UOM_ID','SCALAR_ID','VALUE','STATUS','SYMBOL','TERMINATED','DECIMALS'], axis=1)

# Normalization
# EmployeesMeatPlants_Ontario['NormalizationIndex'] = (EmployeesMeatPlants_Ontario['Employees']-EmployeesMeatPlants_Ontario['Employees'].min())/(EmployeesMeatPlants_Ontario['Employees'].max()-EmployeesMeatPlants_Ontario['Employees'].min())
EmployeesMeatPlants_Quebec['NormalizationIndex(%)'] = (EmployeesMeatPlants_Quebec['Employees']/EmployeesMeatPlants_Quebec['Employees'].sum())*100

# Including those census diviosns with no facilities
CensusDivisions = pd.read_csv('../Datasets/GeodataFiles/CensusDivisions/georef-canada-census-division.csv', sep=";", header=0)
CensusDivisionsQuebec = CensusDivisions[CensusDivisions['Official Code Province / Territory'] == 24].reset_index(drop=True)
CensusDivisionsQuebec_CD=CensusDivisionsQuebec[['Official Code Census Division']].astype(str).rename(columns={'Official Code Census Division':'CD'})

EmployeesMeatPlants_Quebec = EmployeesMeatPlants_Quebec.reset_index()
EmployeesMeatPlants_Quebec = EmployeesMeatPlants_Quebec.merge(CensusDivisionsQuebec_CD, on='CD', how='outer', sort=True).fillna(0)

EmployeesMeatPlants_Quebec = EmployeesMeatPlants_Quebec.set_index('CD')

EmployeesMeatPlants_Quebec.to_csv('../Datasets/AgriProductsProduction/Meat/EmployeesMeatPlants/2021/FederallyLicensedMeatPlants_Quebec.csv')

EmployeesMeatPlants_Quebec

Employees  NormalizationIndex(%)
CD                                    
2401        0.0               0.000000
2402        0.0               0.000000
2403        0.0               0.000000
2404        0.0               0.000000
2405        0.0               0.000000
2406        0.0               0.000000
2407        0.0               0.000000
2408        0.0               0.000000
2409        0.0               0.000000
2410        0.0               0.000000
2411        0.0               0.000000
2412        0.0               0.000000
2413        0.0               0.000000
2414        0.0               0.000000
2415        0.0               0.000000
2416        0.0               0.000000
2417        0.0               0.000000
2418        0.0               0.000000
2419      714.5               9.185576
2420        0.0               0.000000
2421        0.0               0.000000
2422        0.0               0.000000
2423      140.5               1.806261
2425       89.0               1.144179
2426        0.0               0.000000
2427        0.0               0.000000
2428        0.0               0.000000
2429        0.0               0.000000
2430        0.0               0.000000
2431        0.0               0.000000
2432        0.0               0.000000
2433        0.0               0.000000
2434        0.0               0.000000
2435        0.0               0.000000
2436        0.0               0.000000
2437      149.5               1.921964
2438      109.0               1.401298
2439        0.0               0.000000
2440        0.0               0.000000
2441        0.0               0.000000
2442        2.5               0.032140
2443        0.0               0.000000
2444        0.0               0.000000
2445       49.0               0.629942
2446        0.0               0.000000
2447        0.0               0.000000
2448        0.0               0.000000
2449        0.0               0.000000
2450        0.0               0.000000
2451        0.0               0.000000
2452        2.5               0.032140
2453        0.0               0.000000
2454        0.0               0.000000
2455        0.0               0.000000
2456      349.5               4.493154
2457        7.0               0.089992
2458      352.0               4.525294
2459        0.0               0.000000
2460       34.5               0.443530
2461        0.0               0.000000
2462        0.0               0.000000
2463      349.5               4.493154
2464      275.5               3.541814
2465     1712.0              22.009385
2466     2258.5              29.035161
2467      112.5               1.446294
2468        0.0               0.000000
2469        0.0               0.000000
2470        0.0               0.000000
2471       74.5               0.957768
2472      352.0               4.525294
2473      513.5               6.601530
2474        5.0               0.064280
2475        0.0               0.000000
2476        0.0               0.000000
2477        0.0               0.000000
2478        0.0               0.000000
2479        0.0               0.000000
2480        0.0               0.000000
2481        0.0               0.000000
2482        0.0               0.000000
2483        0.0               0.000000
2484        0.0               0.000000
2485        0.0               0.000000
2486        0.0               0.000000
2487        0.0               0.000000
2488        0.0               0.000000
2489        0.0               0.000000
2490        0.0               0.000000
2491        0.0               0.000000
2492        0.0               0.000000
2493        0.0               0.000000
2494      126.0               1.619850
2495        0.0               0.000000
2496        0.0               0.000000
2497        0.0               0.000000
2498        0.0               0.000000
2499        0.0               0.000000

In [51]:
# We convert everyting to multirow
iterables = [AnimalsTotal.columns.to_list(), AnimalsTotal.index.to_list()]
multiindex_AnimalsTotal_CD = pd.MultiIndex.from_product(iterables, names=["Year", "Animal"])


AnimalsTotal_CD = pd.DataFrame(index=['Total Quebec'], columns=multiindex_AnimalsTotal_CD)
for i in AnimalsTotal.columns.to_list():
    for ii in AnimalsTotal.index.to_list():
        AnimalsTotal_CD.loc['Total Quebec',(i,ii)] = AnimalsTotal.loc[ii,i]
        
# Adding Census divisons
for i in EmployeesMeatPlants_Quebec.index.to_list():
    AnimalsTotal_CD.loc[i,:] = AnimalsTotal_CD.loc['Total Quebec',:]*(EmployeesMeatPlants_Quebec.loc[i,'NormalizationIndex(%)']/100)

AnimalsTotal_CD.index.name = 'CD'       

AnimalsTotal_CD.to_csv('../Datasets/AgriProductsProduction/Meat/Conditioned/AnimalsTotal_CD.csv')

AnimalsTotal_CD

Year            2021 (heads)                                                                                                      
Animal                  Hogs         Sheep         Goat         Calves        Cattle          Chicken         Turkey        Rabbit
CD                                                                                                                                
Total Quebec       6932292.0       75517.0       5757.0  152974.025974  43302.411874      185710042.0     3430198.01       90167.0
2401                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2402                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2403                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2404                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2405                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2406                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2407                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2408                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2409                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2410                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2411                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2412                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2413                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2414                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2415                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2416                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2417                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2418                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2419           636770.924214   6936.671145   528.813589   14051.544843    3977.57579  17058536.351353    315083.4323   8282.357974
2420                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2421                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2422                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2423            125215.27621   1364.034004   103.986437    2763.109938    782.154512   3354407.777978   61958.323636   1628.651218
2425            79317.861799     864.05001    65.870412    1750.297398    495.457306   2124856.172527   39247.621378   1031.672302
2426                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2427                     0.0           0.0          0.0            0.0           0.0              0.0            0.0           0.0
2428                     0.0           0.0          0.0            0.0

### Carcass weight datasets

In [52]:
LivetoCarcassWeight_ratio = pd.read_csv('../Datasets/AgriProductsProduction/Meat/LiveCarcassWeightRatio_Conditioned.csv', sep=",", header=0).set_index('Animal')
LivetoCarcassWeight_ratio

Carcass_Live_ratio (%)
Animal                         
Cattle                       49
Hogs                         80
Calves                       55
Sheep                        46
Goat                         50
Rabbit                       53
Chicken                      72
Turkey                       77

In [59]:
CarcassWeight = pd.read_csv('../Datasets/AgriProductsProduction/Meat/WeightCarcasses_Conditioned.csv', sep=",", header=0)
CarcassWeight = CarcassWeight.rename(columns={'Unnamed: 0':'Animals'}).set_index('Animals')
CarcassWeight.loc['Chicken'] = LivetoCarcassWeight_ratio.loc['Chicken'].item()/100*SlaughterChickenTotal.loc['ChickenWeight'].drop(columns=list(set(SlaughterChickenTotal.columns.to_list()) - set(AnimalsTotal.columns.to_list())))
CarcassWeight.loc['Turkey'] = LivetoCarcassWeight_ratio.loc['Turkey'].item()/100*SlaughterTurkeyTotal.loc['TurkeyWeight'].drop(columns=list(set(SlaughterTurkeyTotal.columns.to_list()) - set(AnimalsTotal.columns.to_list())))

CarcassWeight.to_csv('../Datasets/AgriProductsProduction/Meat/WeightCarcasses_Final.csv')

CarcassWeight

2021
Animals            
Cattle   372.841744
Hogs     112.639976
Calves   161.476391
Sheep     19.700000
Goat      14.190000
Rabbit     1.250000
Chicken    1.710516
Turkey     6.971658

### Animal Composition Datasets

In [60]:
AnimalComposition = pd.read_csv('../Datasets/AgriProductsProduction/Meat/Animal compositions/AnimalCompositions.csv', sep=",", header=0).set_index('Animal')
AnimalComposition.columns
AnimalComposition

P (%live_wt)                                                Ref
Animal                                                                  
Cattle          0.740  Standardised calculation methods for\nanimal m...
Hogs            0.532  Standardised calculation methods for\nanimal m...
Calves          0.800  Standardised calculation methods for\nanimal m...
Sheep           0.520  Standardised calculation methods for\nanimal m...
Goat            0.630  Standardised calculation methods for\nanimal m...
Rabbit          0.600  Standardised calculation methods for\nanimal m...
Chicken         0.500  Standardised calculation methods for\nanimal m...
Turkey          0.510  Standardised calculation methods for\nanimal m...

### Final dataset

In [63]:
AnimalsTotalCD_Final = AnimalsTotal_CD.copy()
AnimalsTotalCD_Final = AnimalsTotalCD_Final.drop('Total Quebec', axis=0)

AnimalsTotalCD_index = AnimalsTotal_CD.drop('Total Quebec', axis=0).index.to_list()
# AnimalsTotalCD_columns = [i[0].split(' (heads)')[0] for i in AnimalsTotal_CD.columns.to_list()]
AnimalsTotalCD_columns = AnimalsTotal_CD.drop('Total Quebec', axis=0).columns.to_list()

for i in AnimalsTotalCD_index:
    for ii in AnimalsTotalCD_columns:
        AnimalsTotalCD_Final.loc[i,(ii[0].split(' (heads)')[0]  +' (kg live)',ii[1])] = (AnimalsTotalCD_Final.loc[i,ii]*CarcassWeight.loc[ii[1],ii[0].split(' (heads)')[0]]/
                                                              (LivetoCarcassWeight_ratio.loc[ii[1],'Carcass_Live_ratio (%)']/100))
        
        AnimalsTotalCD_Final.loc[i,(ii[0].split(' (heads)')[0]  +' (kg carcass)',ii[1])] = (AnimalsTotalCD_Final.loc[i,(ii[0].split(' (heads)')[0]   +' (heads)',ii[1])]*
                                                                                          CarcassWeight.loc[ii[1],ii[0].split(' (heads)')[0]])
        
        AnimalsTotalCD_Final.loc[i,(ii[0].split(' (heads)')[0]  +' (kg slaugther waste)',ii[1])] = (AnimalsTotalCD_Final.loc[i,(ii[0].split(' (heads)')[0]   +' (kg live)',ii[1])]-
                                                                 AnimalsTotalCD_Final.loc[i,(ii[0].split(' (heads)')[0]   +' (kg carcass)',ii[1])])
        
        AnimalsTotalCD_Final.loc[i,(ii[0].split(' (heads)')[0]  +' (kg P live)',ii[1])] = (AnimalsTotalCD_Final.loc[i,(ii[0].split(' (heads)')[0]  +' (kg live)',ii[1])]*
                                                                          AnimalComposition.loc[ii[1],'P (%live_wt)']/100)
        
        AnimalsTotalCD_Final.loc[i,(ii[0].split(' (heads)')[0] +' (kg P carcass)',ii[1])] = (AnimalsTotalCD_Final.loc[i,(ii[0].split(' (heads)')[0] +' (kg carcass)',ii[1])]*
                                                                          AnimalComposition.loc[ii[1],'P (%live_wt)']/100)
        
        AnimalsTotalCD_Final.loc[i,(ii[0].split(' (heads)')[0] +' (kg P slaugther waste)',ii[1])] = (AnimalsTotalCD_Final.loc[i,(ii[0].split(' (heads)')[0] +' (kg slaugther waste)',ii[1])]*
                                                                          AnimalComposition.loc[ii[1],'P (%live_wt)']/100)
                
        # AnimalsTotal_Final.loc[i,ii+' (kg slaugther waste)'] = (AnimalsTotal_Final.loc[i,ii +' (kg live)'] -
        #                                                         AnimalsTotal_Final.loc[i,ii +' (kg bones)'] - 
        #                                                         AnimalsTotal_Final.loc[i,ii +' (kg meat)'] -
        #                                                         AnimalsTotal_Final.loc[i,ii +' (kg fat)'] -
        #                                                         AnimalsTotal_Final.loc[i,ii +' (kg blood)'])
AnimalsTotalCD_Final.to_csv('../Datasets/AgriProductsProduction/Meat/Conditioned/AnimalsTotalCD_Final.csv')
AnimalsTotalCD_Final.to_csv('../Results/AgriProducts/P_MeatQuebecRaw.csv')


AnimalsTotalCD_Final

Year      2021 (heads)                                                                                                      2021 (kg live) 2021 (kg carcass) 2021 (kg slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg live) 2021 (kg carcass) 2021 (kg slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg live) 2021 (kg carcass) 2021 (kg slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg live) 2021 (kg carcass) 2021 (kg slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg live) 2021 (kg carcass) 2021 (kg slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg live) 2021 (kg carcass) 2021 (kg slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg live) 2021 (kg carcass) 2021 (kg slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste)  \
Animal            Hogs         Sheep         Goat        Calves        Cattle          Chicken         Turkey        Rabbit           Hogs              Hogs                      Hogs             Hogs                Hogs                        Hogs          Sheep             Sheep                     Sheep            Sheep               Sheep                       Sheep           Goat              Goat                      Goat             Goat                Goat                        Goat         Calves            Calves                    Calves           Calves              Calves                      Calves         Cattle            Cattle                    Cattle           Cattle              Cattle                      Cattle        Chicken           Chicken                   Chicken          Chicken             Chicken                     Chicken         Turkey            Turkey                    Turkey           Turkey              Turkey                      Turkey   
CD                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
2401               0.0           0.0          0.0           0.0           0.0              0.0            0.0           0.0   0.000000e+00      0.000000e+00              0.000000e+00     0.000000e+00        0.000000e+00                    0.000000       0.000000          0.000000                  0.000000         0.000000            0.000000                    0.000000       0.000000          0.000000                  0.000000         0.000000            0.000000                    0.000000   0.000000e+00      0.000000e+00              0.000000e+00         0.000000            0.000000                    0.000000   0.000000e+00      0.000000e+00              0.000000e+00         0.000000            0.000000                    0.000000   0.000000e+00      0.000000e+00              0.000000e+00         0.000000            0.000000                    0.000000   0.000000e+00      0.000000e+00              0.000000e+00         0.000000            0.000000                    0.000000   
2402               0

In [64]:
AnimalsTotal_FinalClean = AnimalsTotalCD_Final[AnimalsTotalCD_Final.filter(regex='kg P').columns].copy()

AnimalsTotal_FinalClean.to_csv('../Results/AgriProducts/P_MeatQuebecClean.csv')

AnimalsTotal_FinalClean

Year   2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste)
Animal             Hogs                Hogs                        Hogs            Sheep               Sheep                       Sheep             Goat                Goat                        Goat           Calves              Calves                      Calves           Cattle              Cattle                      Cattle          Chicken             Chicken                     Chicken           Turkey              Turkey                      Turkey           Rabbit              Rabbit                      Rabbit
CD                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
2401       0.000000e+00        0.000000e+00                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000
2402       0.000000e+00        0.000000e+00                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000
2403       0.000000e+00        0.000000e+00                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000
2404       0.000000e+00        0.000000e+00                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000
2405       0.000000e+00        0.000000e+00                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000        

In [65]:
AnimalsTotalCD_CD2021_aux = pd.read_csv('../Results/AgriProducts/P_MeatQuebecClean.csv', sep=",",header=None, index_col=[0])

for name, values in AnimalsTotalCD_CD2021_aux.iteritems():
    AnimalsTotalCD_CD2021_aux.loc['Name', name] = AnimalsTotalCD_CD2021_aux.loc['Animal', name]+' '+AnimalsTotalCD_CD2021_aux.loc['Year', name]

AnimalsTotalCD_CD2021_aux = AnimalsTotalCD_CD2021_aux.rename(columns=dict(zip(AnimalsTotalCD_CD2021_aux.columns.to_list(),AnimalsTotalCD_CD2021_aux.loc['Name',:].to_list())))
AnimalsTotalCD_CD2021_aux = AnimalsTotalCD_CD2021_aux.drop(['Year','Animal', 'CD', 'Name'], axis=0)
AnimalsTotalCD_CD2021_aux.index.names = ['CD']

AnimalsTotalCD_CD2021_aux

/tmp/ipykernel_8243/89430428.py:3: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for name, values in AnimalsTotalCD_CD2021_aux.iteritems():


Hogs 2021 (kg P live) Hogs 2021 (kg P carcass) Hogs 2021 (kg P slaugther waste) Sheep 2021 (kg P live) Sheep 2021 (kg P carcass) Sheep 2021 (kg P slaugther waste) Goat 2021 (kg P live) Goat 2021 (kg P carcass) Goat 2021 (kg P slaugther waste) Calves 2021 (kg P live) Calves 2021 (kg P carcass) Calves 2021 (kg P slaugther waste) Cattle 2021 (kg P live) Cattle 2021 (kg P carcass) Cattle 2021 (kg P slaugther waste) Chicken 2021 (kg P live) Chicken 2021 (kg P carcass) Chicken 2021 (kg P slaugther waste) Turkey 2021 (kg P live) Turkey 2021 (kg P carcass) Turkey 2021 (kg P slaugther waste) Rabbit 2021 (kg P live) Rabbit 2021 (kg P carcass) Rabbit 2021 (kg P slaugther waste)
CD                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
2401                   0.0                      0.0                              0.0                    0.0                       0.0                               0.0                   0.0                      0.0                              0.0                     0.0                        0.0                                0.0                     0.0                        0.0                                0.0                      0.0                         0.0                                 0.0                     0.0                        0.0                                0.0                     0.0                        0.0                                0.0
2402                   0.0                      0.0                              0.0                    0.0                       0.0                               0.0                   0.0                      0.0                              0.0                     0.0                        0.0                                0.0                     0.0                        0.0                                0.0                      0.0                         0.0                                 0.0                     0.0                        0.0                                0.0                     0.0                        0.0                                0.0
2403                   0.0                      0.0                              0.0                    0.0                       0.0                               0.0                   0.0                      0.0                              0.0                     0.0                        0.0                                0.0                     0.0                        0.0                                0.0                      0.0                         0.0                                 0.0                     0.0                        0.0                                0.0                     0.0                        0.0                                0.0
2404                   0.0                      0.0                              0.0                    0.0                       0.0                               0.0                   0.0                      0.0                              0.0                     0.0                        0.0                                0.0                     0.0                        0.0                                0.0                      0.0                         0.0                                 0.0                     0.0                        0.0                                0.0      

In [66]:
years_complete = [2021]# np.arange(2001,2021+1,1)
years_complete.sort()

# EggProductionOntario_CD2001to2021 = pd.DataFrame(np.nan, index=range(len(EggProductionOntario_CD.CD)*len(years_complete)), columns=['DGUID'])
DGUID_list = []
CD_list = []
REF_DATE_list = []
Animals_list = []
for CD in AnimalsTotalCD_CD2021_aux.index:
    for year in years_complete:
        for animal in ['Hogs','Sheep','Goat','Calves','Cattle','Chicken','Turkey','Rabbit']:
            DGUID_list.append(str(year)+'A0003'+str(CD))
            CD_list.append(CD)
            REF_DATE_list.append(year)
            Animals_list.append(animal)
        
AnimalsTotalCD_CD2021 = pd.DataFrame({'DGUID':DGUID_list,
                                            'CD':CD_list,
                                            'REF_DATE':REF_DATE_list,
                                            'Animal':Animals_list})

AnimalsTotalCD_CD2021 = AnimalsTotalCD_CD2021.sort_values(by=['CD', 'REF_DATE'])


for index, row in AnimalsTotalCD_CD2021.iterrows():
    year = str(AnimalsTotalCD_CD2021.loc[index,'REF_DATE'])
    CenDiv = AnimalsTotalCD_CD2021.loc[index,'CD']
    Animal = AnimalsTotalCD_CD2021.loc[index,'Animal']
    
    label1 = Animal + ' '+ year + ' (kg P live)'
    label2 = Animal + ' '+ year + ' (kg P carcass)'
    label3 = Animal + ' '+ year + ' (kg P slaugther waste)'
    
    AnimalsTotalCD_CD2021.loc[index,'kg P live'] = AnimalsTotalCD_CD2021_aux.loc[CenDiv,label1]
    AnimalsTotalCD_CD2021.loc[index,'kg P carcass'] = AnimalsTotalCD_CD2021_aux.loc[CenDiv,label2]
    AnimalsTotalCD_CD2021.loc[index,'kg P slaugther waste'] = AnimalsTotalCD_CD2021_aux.loc[CenDiv,label3]

AnimalsTotalCD_CD2021 = AnimalsTotalCD_CD2021.set_index('DGUID')
AnimalsTotalCD_CD2021.to_csv('../Results/AgriProducts/P_MeatQuebecClean_2021.csv')
AnimalsTotalCD_CD2021

CD  REF_DATE   Animal kg P live kg P carcass kg P slaugther waste
DGUID                                                                             
2021A00032401  2401      2021     Hogs       0.0          0.0                  0.0
2021A00032401  2401      2021    Sheep       0.0          0.0                  0.0
2021A00032401  2401      2021     Goat       0.0          0.0                  0.0
2021A00032401  2401      2021   Calves       0.0          0.0                  0.0
2021A00032401  2401      2021   Cattle       0.0          0.0                  0.0
...             ...       ...      ...       ...          ...                  ...
2021A00032499  2499      2021   Calves       0.0          0.0                  0.0
2021A00032499  2499      2021   Cattle       0.0          0.0                  0.0
2021A00032499  2499      2021  Chicken       0.0          0.0                  0.0
2021A00032499  2499      2021   Turkey       0.0          0.0                  0.0
2021A00032499  2499      2021   Rabbit       0.0          0.0                  0.0

[784 rows x 6 columns]

In [67]:
AnimalsTotal_FinalSummary = AnimalsTotal_FinalClean[AnimalsTotal_FinalClean.filter(regex='2021').columns].copy()
AnimalsTotal_FinalSummary.to_csv('../Results/AgriProducts/P_MeatQuebecSummary2021.csv')

AnimalsTotal_FinalSummary

Year   2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste) 2021 (kg P live) 2021 (kg P carcass) 2021 (kg P slaugther waste)
Animal             Hogs                Hogs                        Hogs            Sheep               Sheep                       Sheep             Goat                Goat                        Goat           Calves              Calves                      Calves           Cattle              Cattle                      Cattle          Chicken             Chicken                     Chicken           Turkey              Turkey                      Turkey           Rabbit              Rabbit                      Rabbit
CD                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
2401       0.000000e+00        0.000000e+00                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000
2402       0.000000e+00        0.000000e+00                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000
2403       0.000000e+00        0.000000e+00                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000
2404       0.000000e+00        0.000000e+00                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000
2405       0.000000e+00        0.000000e+00                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000                    0.000000         0.000000            0.000000        